### ISSUE TITLE
Support multiple providers

### ISSUE THREAD

**Issue by vdsh** [stefanocudini/leaflet-search#81]
Hi,

For my project, I needed to be able to retrieve information from different content providers. I therefore had to hack around leaflet-search.js and change quite a lot of things. Anyway, it's working fine, I may have missed some things, but it should not be too hard to fine-tune the missing things.

You set-up providers like this:
```javascript
var googleProvider = { 'key': 'google', 'type': 'callData', 'callData': googleGeocoding, 'filterData': filterJSONGoogle, 'order': 2 };
    var nominatimProvider = { 'key': 'nominatim', 'type': 'url', 'url': nominatimUrl, 'filterData': filterJSONNominatim, 'order': 1 };
```
The key allows to track active request (plus to customize the tip based on the key), the type allows to define how to retrieve the information (with callData or url as additionnal parameter) and filterData is the function which takes care of handling the data to return a valid object.

I had to change the structure of object returned to be able to handle those multiple providers. Instead of being {"name": loc[a,b]}, it is now {'title', 'loc', 'type', 'order'}. We are not using the associative arrays because it's not best practice javascript + how to handle if the 2 providers send us the same array key as title?

title being the former "name", 'loc' the same location, type being the provider key (if you want to use this attribute to customize the tip) and order if you want to order the results from the different providers. Here, Nominatim results will always appear above Google's (even if they are retrieved a few seconds after)

Then to add the control, it's really straightforward:
```javascript
 map.addControl( new L.Control.Search({
      providers: { googleProvider, nominatimProvider },
      markerLocation: false,
      autoType: false,
      autoCollapse: true,
      animateLocation: false,		
      circleLocation: false,
      minLength: 2,
      zoom: 10,
      callTip: customTip,
    }) );
```

As it is a rewrite of about half of the functions changing the parameters, it will not be compatible with the current version.
However, updating it is really straightforward. If you need to support multiple providers, I am more than happy to share the changes I did + the example file. Unfortunately, I don't know how I can attach those 2 files.

Cheers

**Comment by stefanocudini** [comment]
Try this for multiple layers..
http://labs.easyblog.it/maps/leaflet-search/examples/multiple-layers.html

**Comment by vdsh** [comment]
Hi,

Thank you for answering. But I don't think multiple layers would fit my usecase, as I want the user to be able either to search a location (e.g a country or a city) or to search directly for some name of shops in my database.

Therefore, I modified the leaflet-search in order to be able to search into multiple providers: google (for the geocoding) plus my application (for the database of shops)

I just wanted to share what I did here as I thought it may help others.

Cheers
